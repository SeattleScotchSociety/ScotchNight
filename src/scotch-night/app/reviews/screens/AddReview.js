// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
    ScrollView,
    View,
    Picker,
    TextInput,
    Text,
    Slider,
    StyleSheet,
    StatusBar
} from 'react-native';
import { Constants } from 'expo';
import * as reviewActions from '../ReviewActions';
import GrowingTextInput from '../../components/GrowingTextInput';

export class AddReview extends Component {
    static navigationOptions = {
        title: 'Review'
    };

    constructor(props) {
        super(props);

        this.state = {
            review: {
                bottleId: '',
                member: '',
                rating: '',
                notes: ''
            }
        };

        this._handleOnAddPress = this._handleOnAddPress.bind(this);
    }

    _handleTextChange = (property, value) => {
        let { review } = this.state;

        review[property] = value;

        this.setState({ review });
    };

    _handleOnAddPress = () => {
        let { addReview } = this.props.actions;
        const { navigate } = this.props.navigation;
        let { review } = this.state;

        addReview(review);
        navigate('BottleList', {});
    };

    _renderBottleItem(bottle) {
        return (
            <Picker.Item
                key={bottle.id}
                label={`${bottle.distillery} ${bottle.name}`}
                value={bottle.id}
            />
        );
    }

    render() {
        let { bottles } = this.props;
        let { review } = this.state;

        return (
            <View style={{ flex: 1 }}>
                <ScrollView
                    style={{ flex: 1, backgroundColor: '#F8F8F9' }}
                    contentContainerStyle={{ paddingTop: 65 + 30 }}>
                    <Picker
                        selectedValue={review.bottleId}
                        onValueChange={id => this._handleTextChange('bottleId', id)}>
                        {bottles.map(bottle => this._renderBottleItem(bottle))}
                    </Picker>
                    <View style={[styles.row, styles.firstRow]}>
                        <TextInput
                            value={this.state.inputValue}
                            onChangeText={this._handleTextChange}
                            style={styles.textInput}
                        />
                    </View>
                    <View style={styles.row}>
                        <TextInput
                            value={this.state.inputValue}
                            onChangeText={this._handleTextChange}
                            style={styles.textInput}
                        />
                    </View>
                    <View style={styles.row}>
                        <TextInput
                            value={this.state.inputValue}
                            onChangeText={this._handleTextChange}
                            style={styles.textInput}
                        />
                    </View>
                    <View style={styles.row}>
                        <GrowingTextInput
                            minHeight={80}
                            value={this.state.inputValue}
                            onChangeText={this._handleTextChange}
                            style={styles.growingTextInput}
                        />
                    </View>
                    <Slider minimumValue={0} maximumValue={5} step={1} />
                </ScrollView>

                <View style={styles.navbar}>
                    <Text style={styles.headerText}>Review</Text>
                </View>

                <StatusBar barStyle="light-content" />
            </View>
        );
    }
}

function mapStateToProps(state) {
    return state;
}

function mapDispatchToProps(dispatch) {
    return { actions: bindActionCreators(reviewActions, dispatch) };
}

export default connect(mapStateToProps, mapDispatchToProps)(AddReview);

const styles = StyleSheet.create({
    navbar: {
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        height: 65,
        backgroundColor: '#050B7A',
        alignItems: 'center',
        justifyContent: 'center',
        paddingTop: Constants.statusBarHeight
    },
    headerText: {
        fontSize: 18,
        color: '#fff',
        fontWeight: '600'
    },
    row: {
        backgroundColor: '#fff',
        borderBottomWidth: StyleSheet.hairlineWidth,
        borderColor: '#ccc'
    },
    firstRow: {
        borderTopWidth: StyleSheet.hairlineWidth,
        borderColor: '#ccc'
    },
    textInput: {
        flex: 1,
        height: 45,
        paddingHorizontal: 15
    },
    growingTextInput: {
        paddingHorizontal: 15,
        paddingVertical: 15,
        fontSize: 15
    }
});
