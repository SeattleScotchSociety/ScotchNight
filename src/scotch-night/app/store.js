import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { AsyncStorage } from 'react-native';
import { persistStore, autoRehydrate } from 'redux-persist';
import { reducer } from './Reducers';
import rootSaga from './Sagas';
import Bottles from './bottles';
let { loadBottles } = Bottles.Actions;

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
    reducer,
    undefined,
    compose(applyMiddleware(sagaMiddleware), autoRehydrate())
);

sagaMiddleware.run(rootSaga);

persistStore(store, { storage: AsyncStorage }, () => {
    store.dispatch(loadBottles());
}).purge();

export default store;
