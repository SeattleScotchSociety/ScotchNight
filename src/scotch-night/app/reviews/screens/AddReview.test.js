import React from 'react';
import renderer from 'react-test-renderer';
import Store from '../../store';
import AddReview from './AddReview';

test('renders correctly', () => {
    const tree = renderer.create(<AddReview store={Store} />).toJSON();
    expect(tree).toMatchSnapshot();
});
