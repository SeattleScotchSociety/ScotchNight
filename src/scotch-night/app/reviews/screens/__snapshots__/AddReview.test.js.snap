// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    contentContainerStyle={
      Object {
        "paddingTop": 95,
      }
    }
    style={
      Object {
        "backgroundColor": "#F8F8F9",
        "flex": 1,
      }
    }
  >
    <View
      style={undefined}
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
          },
          Object {
            "borderColor": "#ccc",
            "borderTopWidth": 0.5,
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "flex": 1,
            "height": 45,
            "paddingHorizontal": 15,
          }
        }
        value={undefined}
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "flex": 1,
            "height": 45,
            "paddingHorizontal": 15,
          }
        }
        value={undefined}
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "flex": 1,
            "height": 45,
            "paddingHorizontal": 15,
          }
        }
        value={undefined}
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
        }
      }
    >
      <TextInput
        minHeight={80}
        multiline={true}
        onChange={[Function]}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        style={
          Array [
            Object {
              "fontSize": 15,
              "paddingHorizontal": 15,
              "paddingVertical": 15,
            },
            Object {
              "height": 80,
            },
          ]
        }
        underlineColorAndroid="rgba(0, 0, 0, 0)"
        value={undefined}
      />
    </View>
    <RCTSlider
      disabled={false}
      enabled={true}
      maximumValue={5}
      minimumValue={0}
      onChange={undefined}
      onResponderTerminationRequest={[Function]}
      onSlidingComplete={undefined}
      onStartShouldSetResponder={[Function]}
      onValueChange={undefined}
      step={1}
      style={
        Array [
          Object {
            "height": 40,
          },
          undefined,
        ]
      }
      value={0}
    />
  </ScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#050B7A",
        "height": 65,
        "justifyContent": "center",
        "left": 0,
        "paddingTop": 20,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
    >
      Review
    </Text>
  </View>
</View>
`;
